### PR TITLE
[Feature] Support Struct and Map type in Hudi

### PR DESCRIPTION
**Pull request by Smith-Cruise**
## What type of PR is this：
- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by Smith-Cruise** [comment]
run starrocks_be_unittest

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:heart_eyes: **pass** : 16 / 18 (88.89%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|16|18|88.89%|[188, 200]|

File: fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java [hunk 1]
```diff
@@ -175,6 +176,34 @@ public static Type fromHudiType(Schema avroSchema) {
...
+            case RECORD:
+                // Struct type
+                List<Schema.Field> fields = avroSchema.getFields();
+                Preconditions.checkArgument(fields.size() > 0);
+                ArrayList<StructField> structFields = new ArrayList<>(fields.size());
+                for (Schema.Field field : fields) {
+                    String fieldName = field.name();
+                    Type fieldType = fromHudiType(field.schema());
+                    if (fieldType.isUnknown()) {
+                        isConvertedFailed = true;
```
Comment (imay) [inline fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:188]: Why not break here?

Comment (Smith-Cruise) [reply fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:188]: The mind wandered.

**Comment by Smith-Cruise** [comment]
run starrocks_fe_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 16 / 21 (76.19%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/connector/ColumnTypeConverter.java|16|21|76.19%|[160, 188, 189, 201, 202]|

**Review by imay — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java [hunk 2]
```diff
@@ -175,6 +176,36 @@ public static Type fromHudiType(Schema avroSchema) {
...
+                        break;
```
Comment (DorianZheng) [inline fe/fe-core/src/main/java/com/starrocks/connector/ColumnTypeConverter.java:189]: One of the subfield of Struct is not recognized by SR, the others still able to be query right? Seems we don't have to terminate the entire convert procedure.

**Review by DorianZheng — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by Smith-Cruise** [comment]
@Mergify backport branch-2.5

**Pull request merged**